### DISCUSSION TITLE
How would I create a custom pokedex and implement custom pokemon?

### DISCUSSION THREAD

**Discussion by atendev** [wavebeem/pkmn.help#238] [General]
I want to create a new regional dex that will only show the pokemon I plan to implement in my game. Also, how would I go about adding a custom pokemon to the pokedex if that is possible?

**Comment by wavebeem** [comment]
You would need to replace `public/data-pkmn.json` with your own JSON file matching the same structure

https://github.com/wavebeem/pkmn.help/blob/master/public/data-pkmn.json

```json
{
  "name": "venusaur-mega",
  "speciesNames": {
    "ja-Hrkt": "フシギバナ",
    "roomaji": "Fushigibana",
    "ko": "이상해꽃",
    "zh-Hant": "妙蛙花",
    "fr": "Florizarre",
    "de": "Bisaflor",
    "es": "Venusaur",
    "it": "Venusaur",
    "en": "Venusaur",
    "ja": "フシギバナ",
    "zh-Hans": "妙蛙花"
  },
  "formNames": {
    "ja-Hrkt": "メガフシギバナ",
    "ko": "메가이상해꽃",
    "zh-Hant": "超級妙蛙花",
    "fr": "Méga-Florizarre",
    "de": "Mega-Form",
    "es": "Mega-Venusaur",
    "it": "MegaVenusaur",
    "en": "Mega Venusaur",
    "ja": "メガフシギバナ",
    "zh-Hans": "超级妙蛙花"
  },
  "number": 3,
  "hp": 80,
  "attack": 100,
  "defense": 123,
  "spAttack": 122,
  "spDefense": 120,
  "speed": 80,
  "id": "10033",
  "types": ["grass", "poison"],
  "imageType": "sprite",
  "hasShiny": true
}
```

This is the format for an entry

**Reply by atendev** [reply]
I changed the .json file but im getting this error. For more info I just deleted the info in there and pasted the above:

`TypeError: allPokemon.map is not a function

http://localhost:5173/defense/

Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36 Edg/120.0.0.0

{"team_types":"[]","loglevel":"WARN","team_abilities":"[]"}`

**Reply by atendev** [reply]
Adding the brackets fixed this problem

**Reply by atendev** [reply]
Holy Crap man, I just have to say this piece of tech is amazing. I wrote a python script to go through my csv file and format it like you wanted. Copy and paste and now I have a whole pokedex. I just have to go and finish putting the stats and things into my excel spreadsheet and export that to csv. 

You're amazing!!!

**Reply by wavebeem** [reply]
very cool!! please share a link when you're done, I'd love to see it
